ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -455,10 +455,21 @@ static int
 azure_object_size(
   WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, wt_off_t *sizep)
 {
-    WT_UNUSED(file_system);
+    azure_file_system *azure_fs = reinterpret_cast<azure_file_system *>(file_system);
+
     WT_UNUSED(session);
```

Comment:
I know this isn't code that is part of the PR, but for future reference: consider using [maybe_unused](https://en.cppreference.com/w/cpp/language/attributes/maybe_unused) rather than `WT_UNUSED`.

Reply:
is there significant trade offs using `WT_UNUSED` over maybe_unused?

Reply:
`maybe_unused` is provided in C++-17 - it's part of the language. `WT_UNUSED` is a C-style WiredTiger specific macro. I would prefer using functionality that is part of the programming language over a third-party library macro for portability and extensibility.

Reply:
thanks for the clarification! I'll keep this in mind for the future